Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -36,14 +37,22 @@ private SessionPool()
                 _idleSessions = new List<SFSession>();
                 _maxPoolSize = MaxPoolSize;
                 _timeout = Timeout;
+                _busySessionsCounter = new FixedZeroCounter();
             }
         }
 
-        private SessionPool(string connectionString, SecureString password) : this()
+        private SessionPool(string connectionString, SecureString password)
         {
-            ConnectionString = connectionString;
-            Password = password;
-            _allowExceedMaxPoolSize = false; // TODO: SNOW-937190
+            lock (s_sessionPoolLock)
```

Comment:
Such a lock is at the pool manager layer. Here we assume problems which are not entirely for this class. I would remove. Moreover, just one constructor has this lock. Seems weird.

Reply:
I believe this lock is to avoid creating at the same time pool for the same connection string in two separate threads.
You are right that ConnectionPoolManager also creates a lock so it could be removed here. I'm removing.
ConnectionCacheManager does not have such a lock so in SessionPool constructor dedicated to it the lock is required.

Reply:
tl; dr; removed